### ISSUE TITLE
AttributesMappingManager performance issue

### ISSUE THREAD

**Issue by admg** [SolrNet/SolrNet#452]
Code in class AttributesMappingManager in lines 46-47:
`var fieldNameCounts = fields.Select(m => m.Key).GroupBy(m => m);
            if (fieldNameCounts.Count() != fields.Count())`
takes extreamly large amount of time. Parsing 250 documents with this code takes 1200 ms. After removing it takes 700ms.

**Comment by admg** [comment]
The problem is probably with Microsoft.DependencyInjection.SolrNet. It uses AttributesMappingManager:
`services.AddSingleton<IReadOnlyMappingManager, AttributesMappingManager>();`
instead of MemoizingMappingManager

**Comment by xmorera** [comment]
@admg would it be possible for you to use a profiler to identify the exact issue? Or isolate the issue in a sample that could be profiled? I know you provided the line, but if you can package a sample with data that can be used for testing, it would be better.

**Comment by admg** [comment]
@xmorera I discovered that problem is with Microsoft.DependencyInjection.SolrNet as I wrote in previous post. Tommorow I'll try to switch to simple DI and check how performance change.

**Comment by admg** [comment]
I compared 2 methods of DI in same project (ASP.NET CORE 2.1), same request and same result class. Response was 250 documents with about 20 fields, mostly strings.
Request made via object provided by CommonServiceLocator was processed 2 times faster.
Currently using Microsoft.DependencyInjection.SolrNet causes performance problems.
I also tried different MappingManagers from https://github.com/SolrNet/SolrNet/blob/master/Documentation/Mapping.md
but there were no noticable improvements of processing time.
